server/src/main/java/org/eclipse/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -371,6 +371,12 @@ private void checkFileControlUniqueness(FileControlEntryContext ctx, String file
     return addTreeNode(ctx, locality -> new SectionNode(locality, SectionType.LOCAL_STORAGE));
   }
 
+  @Override
+  public List<Node> visitSectionName(SectionNameContext ctx) {
+    return addTreeNode(
+        ctx, locality -> new CodeBlockNameNode(locality, ctx.getText()));
```

Comment:
We also have classes ParagraphNode and ProcedureSectionNode for a similar purpose. Please, either delete or reuse them.

Reply:
This new class only deals with the Section and Paragraph names. The existing classes deal with the code definition block as single entity. With the refactoring it makes sense to deal the name nodes separately.